### PR TITLE
Feature/8801 iot thing group

### LINKED ISSUE #8801: Add IoT Thing Group
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Groups are helpful administrating policies across large numbers of iot things.

### New or Affected Resource(s)

New resource based on:
https://docs.aws.amazon.com/cli/latest/reference/iot/create-thing-group.html

### PR DESCRIPTION
**Pull request by Nilos**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #8801

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):
```release-note
New resource: aws_iot_thing_group
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIotThingGroup_'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSIotThingGroup_ -timeout 120m
=== RUN   TestAccAWSIotThingGroup_base
=== PAUSE TestAccAWSIotThingGroup_base
=== RUN   TestAccAWSIotThingGroup_full
=== PAUSE TestAccAWSIotThingGroup_full
=== RUN   TestAccAWSIotThingGroup_name
=== PAUSE TestAccAWSIotThingGroup_name
=== RUN   TestAccAWSIotThingGroup_tags
=== PAUSE TestAccAWSIotThingGroup_tags
=== RUN   TestAccAWSIotThingGroup_propsAttr
=== PAUSE TestAccAWSIotThingGroup_propsAttr
=== RUN   TestAccAWSIotThingGroup_propsDesc
=== PAUSE TestAccAWSIotThingGroup_propsDesc
=== RUN   TestAccAWSIotThingGroup_propsAll
=== PAUSE TestAccAWSIotThingGroup_propsAll
=== RUN   TestAccAWSIotThingGroup_parent
=== PAUSE TestAccAWSIotThingGroup_parent
=== CONT  TestAccAWSIotThingGroup_base
=== CONT  TestAccAWSIotThingGroup_propsDesc
=== CONT  TestAccAWSIotThingGroup_tags
=== CONT  TestAccAWSIotThingGroup_full
=== CONT  TestAccAWSIotThingGroup_propsAll
=== CONT  TestAccAWSIotThingGroup_parent
=== CONT  TestAccAWSIotThingGroup_propsAttr
=== CONT  TestAccAWSIotThingGroup_name
--- PASS: TestAccAWSIotThingGroup_propsDesc (285.78s)
--- PASS: TestAccAWSIotThingGroup_base (295.13s)
--- PASS: TestAccAWSIotThingGroup_name (354.48s)
--- PASS: TestAccAWSIotThingGroup_tags (391.87s)
--- PASS: TestAccAWSIotThingGroup_propsAttr (418.00s)
--- PASS: TestAccAWSIotThingGroup_propsAll (423.11s)
--- PASS: TestAccAWSIotThingGroup_full (511.82s)
--- PASS: TestAccAWSIotThingGroup_parent (561.00s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       561.091s
```

This is an updated version of #12217

### CODE REVIEW

**Comment by nelg** [comment]
Please can we run the automatic review again, it's expired, and what is holding this up from being merged?

**Comment by tyler-aicradle** [comment]
I would happily do any work to get this across the line, I'm just not sure what that is aside from getting the GH actions to pass. What is necessary to move forward?

**Comment by nelg** [comment]
> I would happily do any work to get this across the line, I'm just not sure what that is aside from getting the GH actions to pass. What is necessary to move forward?

Not sure.  I merged it into a custom build of the AWS provider, along with iot provisioning templates.   Has been working fine in my copy.

File: aws/resource_aws_iot_thing_group.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+func expandIotThingsGroupProperties(l []interface{}) *iot.ThingGroupProperties {
+	m := l[0].(map[string]interface{})
+
+	thingGroupProperties := &iot.ThingGroupProperties{}
+
+	if v, ok := m["attributes"]; ok {
+		thingGroupProperties.AttributePayload = &iot.AttributePayload{
+			Attributes: stringMapToPointers(v.(map[string]interface{})),
```
Comment (nelg) [inline aws/resource_aws_iot_thing_group.go:221]: As per git commit:
commit cd020182d2332d17441ccb602d6820c7eca6e54f
Author: Angie Pinilla <angelinepinilla@gmail.com>
Date:   Wed Apr 14 14:03:50 2021 -0400

    migrate stringMapToPointers usage to expandStringMap


I think you will need to change this to use:  expandStringMap

**Comment by ewbankkit** [comment]
Consolidated along with #10634 into #21799.

**Comment by ewbankkit** [comment]
@Nilos Thanks for the contribution :tada: :clap:.

**Pull request closed without merging**